security_check/main.py

```diff
@@ -44,31 +43,27 @@ def _check_image(image, severity, whitelist):
                          vuln.get('vulnerability'))
             continue
         if _filter_severity(vuln['severity'], severity):
-            unpatched += 1
+            unpatched[vuln['vulnerability']] = vuln
 
     if unpatched:
-        base_unpatched = 0
+        base_unpatched = {}
         img = parsed.get('image_analysis')
         if img:
             base_img_url = img[0]['base_image_url']
             base_image = base_img_url[len('https://'):base_img_url.find('@')]
             base_unpatched = _check_image(base_image, severity, whitelist)
```

Comment:
it looks like this might recursively check all base images up to the root. Is there anything to prevent that?